src/dataset.cpp

```diff
@@ -593,6 +593,87 @@ Dataset rebin(const Dataset &d, const Variable &newCoord) {
   return out;
 }
 
+Dataset histogram(const Variable &var, const Variable &coord) {
+  // TODO Is there are more generic way to find "histogrammable" data, not
+  // specific to (neutron) events? Something like Data::ValueVector, i.e., any
+  // data variable that contains a vector of values at each point?
+  const auto &events = var.get<const Data::Events>();
+  // TODO This way of handling events (and their units) as nested Dataset feels
+  // a bit unwieldy. Would it be a better option to store TOF (or any derived
```

Comment:
The current way has a lot of overhead (a dataset with several variables for every pixel), and this is not negligible for large instruments.

The current example is essentially an array (Variable) of structures (Dataset for each pixel) of arrays (variables for TOF, pulse-time, ... for each event).

What I am suggesting is to simply use a structure (Dataset) of arrays (variables for arrays of TOF, pulse-time, ... for each pixel) of arrays (e.g., TOF for each event). This has two advantages:
- We drastically reduce the number of datasets/variables (and thus overhead from more dynamic allocations, base size, and type-erasure).
- We can optimize for small-size vectors, which may drastically reduce the number of allocations needed, e.g., for TOF vectors (all lists that have few events are stored in the small-size optimized vector).

Regarding encapsulation: The only effect is that an `EventList` has we have it now is not a stand-alone object any more. When in a `Dataset` it would simply be access via a proxy object so this would not be an issue.